mathesar_ui/src/pages/data-explorer/DataExplorerPage.svelte

```diff
@@ -10,11 +10,13 @@
   export let schema: SchemaEntry;
   export let queryManager: QueryManager;
 
+  const routeMeta = meta();
+
   $: ({ query } = queryManager);
 
-  function gotoSchema() {
+  function goBackWithFallbackToSchema() {
     const schemaURL = getSchemaPageUrl(database.name, schema.id);
-    router.goto(schemaURL);
+    router.goto($routeMeta.from || schemaURL);
```

Comment:
> is this the right way to handle back navigation

Maybe? I'm not sure. Basically, I don't think we should ever be doing this because I can't imagine a scenario where I think this type of contrived "back" button would be a good design. I opened #1601 requesting that we remove that button altogether, and in a sync call, Pavish, Kriti, and Ghislaine agreed. In place of that button, we'll have a "View Results" _link_ (not button) which only displays while you're editing a saved exploration.

Because of the slight change in design, I've reverted your changes to this file. @pavish will handle the remaining changes needed within `DataExplorerPage` and that work can be considered independent of your work within this PR.